File: TeamCode/src/main/cpp/sources/Position.cpp
#include "../header/Position.h"

Position::Position(double x, double y, double rotation) {
    this->x = x;
    this->y = y;
    this->rotation = rotation;
}

void Position::set(double x, double y, double rotation) {
    this->x = x;
    this->y = y;
    this->rotation = rotation;
}

Position Position::setX(double x) {
    this->x = x;
    return *this;
}

Position Position::setY(double y) {
    this->y = y;
    return *this;
}

Position Position::setRotation(double rotation) {
    this->rotation = rotation;
    return *this;
}

std::string Position::to_string() {
    return (::to_string(x) + "/" + ::to_string(y) + "/" + ::to_string(rotation));
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/math/Position.java
package org.firstinspires.ftc.teamcode.math;

import androidx.annotation.NonNull;

@SuppressWarnings("UnusedReturnValue")
public class Position {

    public Double x, y, rotation;

    public Position(Double x, Double y, Double rotation) {
        this.x = x;
        this.y = y;
        this.rotation = rotation;
    }

    public void set(Double x, Double y, Double rotation) {
        this.x = x;
        this.y = y;
        this.rotation = rotation;
    }

    public Position setX(Double x) {
        this.x = x;
        return this;
    }

    public Position setY(Double y) {
        this.y = y;
        return this;
    }

    public Position setRotation(Double rotation) {
        this.rotation = rotation;
        return this;
    }

    @NonNull
    public String toString() {
        return x + "/" + y + "/" + rotation;
    }
}
